internal/fsutil/os_detection.go

```diff
@@ -0,0 +1,55 @@
+package fsutil
+
+import (
+	"os"
+	"strings"
+)
+
+type OSInfo struct {
+	Name    string
+	Version string
+}
+
+type Detector interface {
+	HasLinuxFile() bool
+	ReadLinuxFile() (string, error)
+	GetInfo(osReleaseContents string) OSInfo
+}
+
+type Detect struct {
+}
+
+func (d *Detect) HasLinuxFile() bool {
+	finfo, err := os.Stat("/etc/os-release")
+	if err != nil {
+		return false
+	}
+	return !finfo.IsDir() && finfo.Size() > 0
+}
+
+func (d *Detect) ReadLinuxFile() (string, error) {
+	nobodyLikesBytes, err := os.ReadFile("/etc/os-release")
```

Comment:
```suggestion
	return io.ReadAll("/etc/os-release")
```

Reply:
lol sorry- i mean to fix that variable name,  but `ReadAll` looks like the better solution all around - thanks!

Reply:
hmmm  : 

> cannot use io.ReadAll("/etc/os-release") (value of type []byte) as string value in return statement

I stand by my claim that nobody likes bytes.

Reply:
eh, my fault. I think `ioutil` used to have something that read it into a string. I may be confusing that with another function.